_overviews/collections-2.13/conversion-between-option-and-collection.md

```diff
@@ -0,0 +1,60 @@
+title: Conversion Between Option and Collection
+partof: collections-213
+overview-name: Collections
+
+num: 18
+previous-page: conversion-between-java-and-scala-collections
+
+permalink: /overviews/collections-2.13/:title.html
+---
+`Option` can be seen conceptually as a collection that has zero or exactly one element, and can be iterated once or more than one time. `Option` does provide a degree of interoperability with the collection types found in the package `scala.collection`, in particular, it implements the interface `IterableOnce`, which models the simplest form of collections. However `Option` is not a full collections in the sense of implementing the full `Iterable` interface, because although an iterator of zero to one elements exists for `Option`, we cannot provide a sensible implementation for [`fromSpecific`](https://github.com/scala/scala/blob/6c68c2825e893bb71d6dc78465ac8c6f415cbd93/src/library/scala/collection/Iterable.scala#L173) which turns one collection into another
+
+
+```
+def fromSpecific(coll: IterableOnce[A]): C
+```
+
+For example if we attempted to defined it as `coll.headOption`, then we would throw away some elements when `coll` has more than one element. For this reason, starting [Scala 2.13](https://github.com/scala/scala/pull/8038), `Option` was made an `IterableOnce` but not an `Iterable`. To help clarify the difference between the two, mentally rename `Iterable` to `Collection` whilst `IterableOnce` becomes `Iterable`.
+
+Hence `Option` can be used everywhere an `IterableOnce` is expected, for example, when calling `flatMap` on a collection (or inside a for-comprehension)
+
+```
+scala> for {
+     |   a <- Set(1)
+     |   b <- Option(41)
+     | } yield (a + b)
+val res0: scala.collection.immutable.Set[Int] = Set(42)
+``` 
+
+since `Set#flatMap` takes function to `IterableOnce`
+
+```
+def flatMap[B](f: Int => scala.collection.IterableOnce[B]): scala.collection.immutable.Set[B]
+```
+
+Having said `Option` is not a full collection in the sense of extending `Iterable`, nevertheless there exists an [implicit conversion](https://github.com/scala/scala/blob/6c68c2825e893bb71d6dc78465ac8c6f415cbd93/src/library/scala/Option.scala#L19) between an `Option` and `Iterable` (or `Collection` in our mental model)
+
+
+```
+implicit def option2Iterable[A](xo: Option[A]): Iterable[A]
+```
+
+so although `Option[A]` is not a full collection it can be _viewed_ as one. For example,
+
+```
+scala> Some(42).drop(1)
+val res0: Iterable[Int] = List()
+```
+
+expands to
+
+```
+scala> Option.option2Iterable(Some(42)).drop(1)
+val res1: Iterable[Int] = List()
+```
+
+because `drop` is not defined on `Option`. Downside of the above implicit conversion is that instead of getting back specific `Option[A]` we are left with more general `Iterable[A]`. For this reason `Option` scaladoc carries the following [note](https://github.com/scala/scala/blob/6c68c2825e893bb71d6dc78465ac8c6f415cbd93/src/library/scala/Option.scala#L124):
```

Comment:
```suggestion
because `drop` is not defined on `Option`. A downside of the above implicit conversion is that instead of getting back an `Option[A]` we are left with an `Iterable[A]`. For this reason, `Option`’s documentation carries the following note:
```